src/java.base/unix/native/libjli/java_md.c

```diff
@@ -498,6 +506,10 @@ GetJREPath(char *path, jint pathsize, jboolean speculative)
     JLI_TraceLauncher("Attempt to get JRE path from launcher executable path\n");
 
     if (GetApplicationHome(path, pathsize)) {
+        if (JLI_IsStaticallyLinked()) {
```

Comment:
In passing, GetJREPath's function description includes "or registry settings" which is confusing to see now.

Reply:
I updated the description.

Also, the name "JRE" is severly outdated, but cleaning out technical debt in the JLI library will have to wait for another day.